metal/cache.h

```diff
@@ -11,51 +11,26 @@
  */
 #include <stdint.h>
 
-struct metal_cache;
-
-struct __metal_cache_vtable {
-    void (*init)(struct metal_cache *cache, int ways);
-    int (*get_enabled_ways)(struct metal_cache *cache);
-    int (*set_enabled_ways)(struct metal_cache *cache, int ways);
-};
-
-/*!
- * @brief a handle for a cache
- */
-struct metal_cache {
-    const struct __metal_cache_vtable *vtable;
-};
-
 /*!
- * @brief Initialize a cache
- * @param cache The handle for the cache to initialize
- * @param ways The number of ways to enable
- *
- * Initializes a cache with the requested number of ways enabled.
+ * @brief Initialize L2 cache controller
+ * @param None
+ * @return 0 If no error
  */
-__inline__ void metal_cache_init(struct metal_cache *cache, int ways) {
-    cache->vtable->init(cache, ways);
-}
+int metal_l2cache_init(void);
 
 /*!
- * @brief Get the current number of enabled cache ways
- * @param cache The handle for the cache
- * @return The current number of enabled cache ways
+ * @brief Get the current number of enabled L2 cache ways
+ * @param None
+ * @return The current number of enabled L2 cache ways
  */
-__inline__ int metal_cache_get_enabled_ways(struct metal_cache *cache) {
-    return cache->vtable->get_enabled_ways(cache);
-}
+int metal_l2cache_get_enabled_ways(void);
```

Comment:
Same here, keep the current API, add a comment it will be deprecated and call the new API